syntax = "proto3";

package masterworker;

service Worker{
    rpc MapReduceWork(MapReduceInfo) returns (MapReduceReply) {}
}

message MapReduceInfo{
    string user_id = 1;
    repeated FileInfo file_infos = 2;
    string buffer_file = 3;
    string output_dir = 4;
    bool is_map_work = 5;
}

message FileInfo{
    string filename = 1;
    int32 start =2;
    int32 end = 3;
}

message MapReduceReply{
    string address = 1; 
    string buffer_file = 2;
}

